xgboost_ray/main.py

```diff
@@ -538,8 +540,12 @@ def _train():
                         "evals_result": evals_result,
                         "train_n": self._local_n[dtrain]
                     })
-            except XGBoostError:
-                # Silent fail, will be raised as RayXGBoostTrainingStopped
+            except EarlyStopException:
+                # Usually this should be caught by XGBoost core.
+                # Silent fail, will be raised as RayXGBoostTrainingStopped.
+                return
+            except XGBoostError as e:
+                error_dict.update({"exception": e})
```

Comment:
would it be possible to add a quick test?

Reply:
Sure, I'll add one tomorrow